testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestAzureBlobFileSystem.java

```diff
@@ -0,0 +1,83 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.hive;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.tempto.AfterTestWithContext;
+import io.trino.tempto.BeforeTestWithContext;
+import io.trino.tempto.ProductTest;
+import io.trino.tempto.assertions.QueryAssert;
+import io.trino.testng.services.Flaky;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.tests.product.TestGroups.AZURE;
+import static io.trino.tests.product.utils.HadoopTestUtils.ERROR_COMMITTING_WRITE_TO_HIVE_ISSUE;
+import static io.trino.tests.product.utils.HadoopTestUtils.ERROR_COMMITTING_WRITE_TO_HIVE_MATCH;
+import static io.trino.tests.product.utils.QueryExecutors.onHive;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class TestAzureBlobFileSystem
+        extends ProductTest
+{
+    private String schemaLocation;
+
+    @BeforeTestWithContext
+    public void setUp()
+    {
+        String container = requireNonNull(System.getenv("ABFS_CONTAINER"), "Environment variable not set: ABFS_CONTAINER");
+        String account = requireNonNull(System.getenv("ABFS_ACCOUNT"), "Environment variable not set: ABFS_ACCOUNT");
+        schemaLocation = format("abfs://%s@%s.dfs.core.windows.net/%s", container, account, "test_" + randomNameSuffix());
+
+        onHive().executeQuery("dfs -rm -f -r " + schemaLocation);
+        onHive().executeQuery("dfs -mkdir -p " + schemaLocation);
+    }
+
+    @AfterTestWithContext
+    public void tearDown()
+    {
+        onHive().executeQuery("dfs -mkdir -p " + schemaLocation);
+    }
+
+    @Test(groups = AZURE)
+    @Flaky(issue = ERROR_COMMITTING_WRITE_TO_HIVE_ISSUE, match = ERROR_COMMITTING_WRITE_TO_HIVE_MATCH)
+    public void testPathContainsSpecialCharacter()
+    {
+        String tableName = "test_path_special_character" + randomNameSuffix();
+        String tableLocation = schemaLocation + "/" + tableName;
+
+        onHive().executeQuery("CREATE TABLE " + tableName + " (id bigint) PARTITIONED BY (part string) LOCATION '" + tableLocation + "'");
+        onHive().executeQuery("INSERT INTO " + tableName + " VALUES (1, 'hive=equal'), (2, 'hive:colon'), (3, 'hive+plus')");
+        onTrino().executeQuery("INSERT INTO " + tableName + " VALUES (11, 'trino=equal'), (12, 'trino:colon'), (13, 'trino+plus')");
```

Comment:
Test case with a space would be good too
```suggestion
        onHive().executeQuery("INSERT INTO " + tableName + " VALUES (1, 'hive=equal'), (2, 'hive:colon'), (3, 'hive+plus'), (4, 'hive space')");
        onTrino().executeQuery("INSERT INTO " + tableName + " VALUES (11, 'trino=equal'), (12, 'trino:colon'), (13, 'trino+plus'), (14, 'trino space')");
```

Reply:
and `%` since it's what is used in url encoding